### PR TITLE
Fix creating cursor image from `AtlasTexture`

### LINKED ISSUE #92671: Custom mouse cursors using AtlasTexture are broken in 4.3 beta1
### Tested versions

- Reproducible in 4.3.beta1
- This worked fine in previous 4.3 alpha versions

### System information

MacOS Sonoma 14.4.1 - Godot v4.3.beta1 - forward_plus - vulkan

### Issue description

In previous 4.3 (and 4.2, etc) builds, I could use an AtlasTexture to set a Texture2D value to be set as a custom mouse cursor with `Input.set_custom_mouse_cursor`. This no longer works, and the entire atlas is rendered on screen as the cursor.

See the attached video showing the problem.

https://github.com/godotengine/godot/assets/2903406/8aadda38-f6e0-489e-8b77-349b26a721be

### Steps to reproduce

1. Set two exported Texture2D values using an AtlasTexture.
2. Use those Texture2D values to set the default custom mouse cursor and the pointing hand custom mouse cursor using `Input.set_custom_mouse_cursor`.
3. Run the project and see that the entire atlas is rendered as the mouse cursor, not just the texture selected within the atlas.

This picture shows the simple repro setup required:

Screenshot 2024-06-02 at 10 40 45 AM

The code to reproduce is also simple:

```
extends Control

@export var default_mouse_cursor: Texture2D
@export var pointing_hand_mouse_cursor: Texture2D


# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	Input.set_custom_mouse_cursor(default_mouse_cursor)
	Input.set_custom_mouse_cursor(pointing_hand_mouse_cursor, Input.CURSOR_POINTING_HAND)
```

### Minimal reproduction project (MRP)

Just run this project to see the behavior
[mouse-cursor-bug-repro.zip](https://github.com/user-attachments/files/15524072/mouse-cursor-bug-repro.zip)

### PR DESCRIPTION
**Pull request by kleonc**
Removes special handling of AtlasTexture when creating cursor Image from it, as `AtlasTexture::get_image` already extracts the region and thus can be used directly.

Fixes #92671 (I've tested only on Windows, please check/test different platforms).

| Before<br>(v4.3.beta1.official [a4f2ea91a])| After<br>(this PR)|
|--------|--------|
|f6WzYkajot|ab5EKqLksw|

### CODE REVIEW

File: servers/display_server.h [hunk 1]
```diff
@@ -558,7 +558,7 @@ class DisplayServer : public Object {
 	virtual int tablet_get_driver_count() const { return 1; };
 	virtual String tablet_get_driver_name(int p_driver) const { return "default"; };
 	virtual String tablet_get_current_driver() const { return "default"; };
-	virtual void tablet_set_current_driver(const String &p_driver){};
+	virtual void tablet_set_current_driver(const String &p_driver) {};
```
Comment (AThousandShips) [inline servers/display_server.h:561]:
```suggestion
	virtual void tablet_set_current_driver(const String &p_driver) {}
```

File: servers/display_server.h [hunk 2]
```diff
@@ -399,7 +399,7 @@ class DisplayServer : public Object {
 	virtual void delete_sub_window(WindowID p_id);
 
 	virtual WindowID window_get_active_popup() const { return INVALID_WINDOW_ID; };
-	virtual void window_set_popup_safe_rect(WindowID p_window, const Rect2i &p_rect){};
+	virtual void window_set_popup_safe_rect(WindowID p_window, const Rect2i &p_rect) {};
```
Comment (AThousandShips) [inline servers/display_server.h:402]:
```suggestion
	virtual void window_set_popup_safe_rect(WindowID p_window, const Rect2i &p_rect) {}
```

File: servers/display_server.h (hunk 1 above)
Comment (kleonc) [reply servers/display_server.h:561]: I guess I have something with auto formatting outdated. But tests seem to suggest reverting it back like:
```suggestion
	virtual void tablet_set_current_driver(const String &p_driver){};
```
So do you know if/which one is preferred?

Comment (AThousandShips) [reply servers/display_server.h:561]: Yes remove the `;`

Comment (kleonc) [reply servers/display_server.h]: Great, indeed this one passes too. Thanks!

**Review by deralmas — APPROVED** [review]
Very nice! I always like negative diffs :)

File: servers/display_server.h [hunk 3]
```diff
@@ -399,7 +399,7 @@ class DisplayServer : public Object {
 	virtual void delete_sub_window(WindowID p_id);
 
 	virtual WindowID window_get_active_popup() const { return INVALID_WINDOW_ID; };
-	virtual void window_set_popup_safe_rect(WindowID p_window, const Rect2i &p_rect){};
```
Comment (KoBeWi) [inline servers/display_server.h:402]: You could remove trailing `;` from other methods too.

File: servers/display_server.cpp [hunk 4]
```diff
@@ -1124,35 +1124,23 @@ void DisplayServer::_bind_methods() {
...
-
-		if (atlas_texture.is_valid()) {
-			texture = atlas_texture->get_atlas();
-			r_atlas_rect.size = texture->get_size();
-			r_atlas_rect.position = atlas_texture->get_region().position;
-			texture_size = atlas_texture->get_region().size;
-		} else {
-			texture_size = texture->get_size();
-		}
+		// `AtlasTexture::get_image` extracts the region, no separate case needed.
```
Comment (KoBeWi) [inline servers/display_server.cpp:1133]: The comment is unnecessary and without context it mentions AtlasTexture for no reason.

**Review by KoBeWi — APPROVED** [review]
The reason why #88970 broke this is that I missed the purpose of the atlas rect. The code had a special case for AtlasTexture, where it extracted the base texture and region separately and then re-created a sub-region image 🤦‍♂️

File: servers/display_server.cpp (hunk 4 above)
Comment (kleonc) [reply servers/display_server.cpp]: Removed, as well as the header include which I haven't done before.

File: servers/display_server.h (hunk 3 above)
Comment (kleonc) [reply servers/display_server.h:402]: Done.;

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**